apps/hash-graph/lib/authorization/src/backend.rs

```diff
@@ -6,22 +6,47 @@ use std::error::Error;
 use error_stack::Report;
 
 pub use self::spicedb::{SpiceDb, SpiceDbConfig};
-use crate::zanzibar::{Affiliation, Consistency, Resource, StringTuple, Subject, Zookie};
+use crate::zanzibar::{Affiliation, Consistency, Relation, Resource, StringTuple, Subject, Zookie};
 
 /// A backend for interacting with an authorization system based on the Zanzibar model.
 pub trait AuthorizationBackend {
-    /// Returns if the [`Subject`] has the specified permission or relation to an [`Resource`].
-    async fn check<S, A, R>(
+    /// Creates a new relation between a [`Subject`] and an [`Resource`] with the specified
+    /// [`Affiliation`].
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the relation already exists or could not be created.
+    async fn create_relation<R, A, S>(
         &self,
-        subject: &S,
+        resource: &R,
         affiliation: &A,
+        subject: &S,
+    ) -> Result<CreateRelationResponse, Report<CreateRelationError>>
+    where
+        R: Resource + ?Sized + Sync,
+        A: Relation<R> + ?Sized + Sync,
```

Comment:
Shouldn't this be `Affiliation` instead?

Reply:
No, you cannot add a permission, only a relation.